packages/rspack-cli/src/utils/crossImport.ts

```diff
@@ -0,0 +1,17 @@
+import { pathToFileURL } from "url";
+import isEsmFile from "./isEsmFile";
+
+const crossImport = async <T = any>(
+	path: string,
+	cwd = process.cwd()
+): Promise<T> => {
+	if (isEsmFile(path, cwd)) {
+		const url = pathToFileURL(path).href;
+		const { default: config } = await import(url);
```

Comment:
this will crash node in jest test due to https://github.com/nodejs/node/issues/35889, and I'm not sure how webpack-cli workaround it, @alexander-akait can help me here?

Reply:
We spawn a new process for the each test in CLI integration tests, so we don't have `jest` in scope, to collect coverage we use `nyc` (yeah, it is deprecated, but works).

Reply:
i dont get it why nyc can workaround this problem here since it still runs jest  and it seems it doesnt enable vm-modules，how can jest run dynamic import without vm-moduels enabled？

Reply:
`nyc` is not a workaround, the main workaround - we spawn a new process inside `it`, i.e. (pseudo code):
```
it("serve command", () => { 
  spawnNewProcess(__dirname, ["serve"]); 
});
```

So we don't have `jest` inside a new spawed process, it also allows you to test as if the user has run it - no polluting, just a fresh process.

`nyc` is a workaround to get coverage after testing, because we don't have `jest` inside a new spawned process we can't collect lines, so here we use `nyc` inside a new spawned process and collect coverage (but we collect it only for CI)